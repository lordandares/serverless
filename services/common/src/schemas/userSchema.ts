import { InferType, boolean, number, object, string } from 'yup'

export const userSchema = object().shape({
  _id: string(),
  anonymous: boolean(),
  auth: object().shape({
    anonymousId: string(),
    failedLoginCount: number(),
    inviteToken: string(),
    password: string(),
    token: string(),
    resetPassword: object().shape({
      expires: string(),
      token: string(),
    }),
    username: string().required(),
  }),
  email: string().required(),
  firstName: string().required(),
  lastName: string().required(),
  location: string().nullable(),
  mobile: string(),
  superadmin: boolean(),
  tempPassword: string(),
})

export type UserSchema = InferType<typeof userSchema>

export default userSchema
