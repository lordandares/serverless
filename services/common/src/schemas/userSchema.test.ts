import { ValidationError } from 'yup'
import { userSchema } from './userSchema'

test('valid user', () => {
  expect.assertions(1)

  const validate = userSchema.validate(
    {
      auth: {
        username: 'joe-bloggs',
      },
      email: 'joe@bloggs.com',
      firstName: 'Joe',
      lastName: 'Bloggs',
    },
    { strict: true },
  )

  return expect(validate).resolves.toMatchSnapshot()
})

test('missing required field', () => {
  expect.assertions(1)

  const validate = userSchema.validate(
    {
      auth: {
        username: 'joe-bloggs',
      },
      email: 'joe@bloggs.com',
      firstName: null,
      lastName: 'Bloggs',
    },
    { strict: true },
  )

  return expect(validate).rejects.toThrowError(ValidationError)
})
