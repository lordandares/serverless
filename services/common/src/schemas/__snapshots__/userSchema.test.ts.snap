// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`valid user 1`] = `
Object {
  "auth": Object {
    "username": "joe-bloggs",
  },
  "email": "joe@bloggs.com",
  "firstName": "Joe",
  "lastName": "Bloggs",
}
`;
